### ISSUE TITLE
Bogus undefined reference in an indented blockquote

### ISSUE THREAD

**Issue by nex3** [tcort/markdown-link-check#255]
Minimal reproduction:

```md
* > [Bar]
  >
  > [Bar]: http://example.org
```

markdown-link-check incorrectly flags "bar" as an undefined reference.

**Comment by nex3** [comment]
Actually, it seems that this may happen with any reference defined in a doubly-nested context:

```md
* * [Bar]
  
    [Bar]: http://example.org
```

**Comment by nex3** [comment]
As far as I know this is still a bug!

**Comment by nex3** [comment]
Still not stale

**Comment by nex3** [comment]
Still not stale

**Comment by nex3** [comment]
would you believe it's still not stale

**Comment by nex3** [comment]
they're good bugs, brent

**Comment by nex3** [comment]
T_T

**Comment by nex3** [comment]
🤡
